addons/google_calendar/models/google_calendar.py

```diff
@@ -647,9 +647,16 @@ def create_new_events(self):
             ('event_id.final_date', '>', fields.Datetime.to_string(self.get_minTime())),
         ])
         for att in my_attendees:
-            other_google_ids = [other_att.google_internal_event_id for other_att in att.event_id.attendee_ids if
-                                other_att.google_internal_event_id and other_att.id != att.id and not other_att.google_internal_event_id.startswith('_')]
-            for other_google_id in other_google_ids:
+            # Get the attendees of the same event with google id set
+            other_attendees = att.event_id.attendee_ids.filtered(lambda other_att: other_att.google_internal_event_id and other_att.id != att.id and not other_att.google_internal_event_id.startswith('_'))
+            if att.partner_id in other_attendees.mapped('partner_id'):
+                # After a contacts merge there may be events
+                # with multiple attendee records of the same partner.
+                # Deleting duplicate attendee to avoid raising error on constraint google_id_uniq
+                att.unlink()
```

Comment:
Are we sure the current user has the right to delete any attendee?

Reply:
any internal user has the right to create/modify/delete at least its own attendees
```
id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
access_calendar_attendee_portal,calendar.attendee_portal,model_calendar_attendee,base.group_portal,0,0,0,0
access_calendar_attendee_employee,calendar.attendee_employee,model_calendar_attendee,base.group_user,1,1,1,1
```
the sync operation is done in the user context, from the calendar view button and from the cron
those attendee will have the same partner_id of the current user because of the search domain
https://github.com/odoo/odoo/blob/dfd3af8fd6c609b7a5d7419a134aa470b58fc41a/addons/google_calendar/models/google_calendar.py#L643